From 8dd58a519f79f0742d4c68fbccb2aed2ddb651e8 Mon Sep 17 00:00:00 2001
From: Snappy Team <no-reply@google.com>
Date: Mon, 24 Jan 2022 09:05:38 +0000
Subject: [PATCH 1/1] Fix compilation for older GCC and Clang versions.

Not everything defining __GNUC__ supports flag outputs
from asm statements; in particular, some Clang versions
on macOS does not. The correct test per the GCC documentation
is __GCC_ASM_FLAG_OUTPUTS__, so use that instead.

PiperOrigin-RevId: 423749308
---
 snappy.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/snappy.cc b/snappy.cc
index ee9a2c4..30505ad 100644
--- a/snappy.cc
+++ b/snappy.cc
@@ -1041,7 +1041,7 @@ size_t AdvanceToNextTagX86Optimized(const uint8_t** ip_p, size_t* tag) {
   size_t literal_len = *tag >> 2;
   size_t tag_type = *tag;
   bool is_literal;
-#if defined(__GNUC__) && defined(__x86_64__)
+#if defined(__GCC_ASM_FLAG_OUTPUTS__) && defined(__x86_64__)
   // TODO clang misses the fact that the (c & 3) already correctly
   // sets the zero flag.
   asm("and $3, %k[tag_type]\n\t"
-- 
2.36.1

